ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/TestBlock.hs

```diff
@@ -75,7 +75,7 @@ parseImpl hasFS@HasFS{..} path =
                -> Word64
                -> m ([(SlotOffset, (BlockSize, BlockInfo BlockId))], Maybe ())
             go ls n = do
-                bs <- hGet hndl binarySize
+                bs <- BL.toStrict <$> hGetLenient hasFS hndl binarySize
```

Comment:
Let's document that we are using `hGetLenient` here as a proxy for detecting end-of-file: we're expecting either exactly `binarySize` or exactly 0.